src/runtime_src/core/pcie/emulation/common_em/config.h

```diff
@@ -89,6 +90,89 @@ namespace xclemulation{
   const int VIVADO_MIN_VERSION = 2000;
   const int VIVADO_MAX_VERSION = 2100;
 
+inline bool file_exists(const std::string& fnm)
+{
+  struct stat statBuf;
+  return stat(fnm.c_str(), &statBuf) == 0;
+}
+
+
+//sparse log utility
+enum class eEmulationType{
+  eSw_Emu,
+  eHw_Emu
+};
+
+struct sParseLog
+{
+  std::ifstream file;
+  std::string mFileName;
+  std::atomic_bool mFileExists;
```

Comment:
As you mentioned std::atomic_bool is a typedef to std::atomic<bool>. It is used here.

Reply:
Not convenient at all compared to std::atomic<bool>,  but your choice.

Reply:
I guess what @stsoe means is that C++11 `std::atomic<bool>` is clear, you just need to know what `std::atomic` and `bool` are. It stores O(n) concepts in our brain. :-)
If you use `std::atomic_bool` which comes from C11 https://en.cppreference.com/w/c/thread#Atomic_operations and it stores O(n2) concepts in our brain. Not template in C, so more brain storage. ;-) Do not use C because my brain is too old. :-)